configs/encapsulate_in_http2_connect.yaml

```diff
@@ -26,6 +26,10 @@ static_resources:
           cluster: "cluster_0"
           tunneling_config:
             hostname: host.com:443
```

Comment:
@lambdai can we also set this hostname using formatter like %REQUESTED_SERVER_NAME% so that the CONNECT uses SNI as host name? IF not, how can I achieve that with this feature?

Reply:
IMO the non-trivial operation should be achieve via the headers_to_add field.

The headers_to_add  type is HeaderValueOption. It includes the override option OVERWRITE_IF_EXISTS_OR_ADD and the implementation is on the go

WDTY?

Reply:
Yes. But I see this comment which says for host it is not supported https://github.com/envoyproxy/envoy/blob/45ec050f91407147ed53a999434b09ef77590177/api/envoy/extensions/filters/network/tcp_proxy/v3/tcp_proxy.proto#L81

Reply:
@lambdai WDYT?